### PR TITLE
[types/binary-split]: Add new type definitions for module

### PR DESCRIPTION
**Pull request by krisztianb**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/binary-split/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// Project: https://github.com/maxogden/binary-split#readme
+// Definitions by: Krisztián Balla <https://github.com/krisztianb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.2
+
+/// <reference types="node" />
+
+import { Transform } from 'stream';
+
+export = binary_split;
```
Comment (peterblazejewicz) [inline types/binary-split/index.d.ts:11]:
```suggestion
export = BinarySplit;
```

File: types/binary-split/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Split streams of binary data.
+ * @param splitOn The matcher for the splitting points in the stream. Default: os.EOL
+ */
+declare function binary_split(
```
Comment (peterblazejewicz) [inline types/binary-split/index.d.ts:17]:
```suggestion
declare function BinarySplit(
```

**Review by peterblazejewicz — COMMENTED** [review]
@krisztianb any specific reason to now use a name based on the JS source?
(I've added suggestions based on that original function name)

**Comment by krisztianb** [comment]
@peterblazejewicz the file was generated using dts-gen. I can change that. Thanks for reviewing.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@krisztianb thx for contribution!

**Pull request merged**